### PR TITLE
Fix scrollview scrollbars

### PR DESCRIPTION
**Pull request by reesscot**
#### Description of changes
This fixes an issue where the ScrollView scrollbars always show despite if there is anything to scroll inside.

Marked the unneeded `orientation` prop as deprecated and removed from docs. I didn't actually remove the prop or styling since that would be a breaking change. The `orientation` prop will be removed in 3.0.

#### Description of how you validated changes
yarn build && yarn docs dev

#### Checklist

- [x] PR description included
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by zchenwei — APPROVED** [review]
LGTM! Do we need a changeset?

**Review by jacoblogan — APPROVED** [review]
LGTM, +1 on changeset for ui package

File: docs/src/pages/components/scrollview/demo.tsx [hunk 1]
```diff
@@ -1,41 +1,28 @@
...
-  width="400px"
->
+export const ScrollViewDemo = () => {
+  const {
+    tokens: { colors },
+  } = useTheme();
+  return (
+    <Demo
+      code={`
+<ScrollView width="100%" height="300px" maxWidth="580px">
```
Comment (0618) [inline docs/src/pages/components/scrollview/demo.tsx:12]: for the `height`, should we use `vb` instead, for responsiveness?

File: docs/src/pages/components/scrollview/demo.tsx [hunk 2]
```diff
@@ -1,41 +1,28 @@
...
-  return (
-    <Demo
-      code={propsToCode(scrollViewProps)}
-      propControls={<ScrollViewPropControls {...scrollViewProps} />}
+</ScrollView>`}
     >
       <ScrollView
-        orientation={scrollViewProps.orientation}
+        width="100%"
         height="300px"
```
Comment (0618) [inline docs/src/pages/components/scrollview/demo.tsx:23]: same here

File: docs/src/pages/components/scrollview/examples/ScrollViewHorizontalExample.tsx [hunk 3]
```diff
@@ -2,11 +2,7 @@ import { ScrollView } from '@aws-amplify/ui-react';
 
 export const ScrollViewHorizontalExample = () => {
   return (
-    <ScrollView
-      width="200px"
-      className="horizontal-example"
-      orientation="horizontal"
-    >
+    <ScrollView width="200px" className="horizontal-example">
```
Comment (0618) [inline docs/src/pages/components/scrollview/examples/ScrollViewHorizontalExample.tsx:5]: and here

File: docs/src/pages/components/scrollview/examples/ScrollViewVerticalExample.tsx [hunk 4]
```diff
@@ -2,7 +2,7 @@ import { ScrollView } from '@aws-amplify/ui-react';
 
 export const ScrollViewVerticalExample = () => {
   return (
-    <ScrollView height="100px" width="200px" orientation="vertical">
+    <ScrollView height="100px" width="200px">
```
Comment (0618) [inline docs/src/pages/components/scrollview/examples/ScrollViewVerticalExample.tsx:5]: here

**Review by 0618 — APPROVED** [review]
Kudos to bias for action 👍

File: docs/src/pages/components/scrollview/demo.tsx [hunk 5]
```diff
@@ -1,41 +1,28 @@
...
-  height="300px"
-  width="400px"
->
+export const ScrollViewDemo = () => {
+  const {
+    tokens: { colors },
+  } = useTheme();
+  return (
+    <Demo
+      code={`
```
Comment (zchenwei) [inline docs/src/pages/components/scrollview/demo.tsx:11]: Can we move the code outside and use a variable to store it?

**Review by zchenwei — APPROVED** [review]

File: docs/src/pages/components/scrollview/demo.tsx (hunk 1 above)
Comment (reesscot) [reply docs/src/pages/components/scrollview/demo.tsx:12]: `vb` is an experimental length property that is not supported by most browsers:
https://developer.mozilla.org/en-US/docs/Web/CSS/length

File: packages/ui/src/theme/css/component/scrollView.scss [hunk 6]
```diff
@@ -1,14 +1,16 @@
 .amplify-scrollview {
   display: block;
-  overflow: scroll;
+  overflow: auto;
 
+  // @TODO: Deprecated - Remove in 3.0
   &[data-orientation='horizontal'] {
     overflow-x: scroll;
```
[the remark this answers is not in the document]
Comment (reesscot) [reply packages/ui/src/theme/css/component/scrollView.scss:7]: Yeah, I didn't mean to change anything but the base overflow, will update.

File: docs/src/pages/components/scrollview/demo.tsx (hunk 5 above)
Comment (reesscot) [reply docs/src/pages/components/scrollview/demo.tsx]: fixed

**Pull request merged**